src/features/library/class-allowed-blocks.php

```diff
@@ -44,12 +44,17 @@ public function boot(): void {
 	 * @param WP_Block_Editor_Context $block_editor_context The current block editor context.
 	 * @return bool|string[] Updated block type slugs.
 	 */
-	public function filter_allowed_block_types( $allowed_block_types, $block_editor_context ) {
+	public function filter_allowed_block_types( $allowed_block_types, $block_editor_context ): array|bool {
 		if ( in_array( $block_editor_context->name, (array) $this->context, true ) ) {
+			/**
+			 * Tell PHPStan the returned value is an array.
+			 *
+			 * @var string[] $updated
+			 */
 			$updated = array_reduce(
```

Comment:
This core function is one of those that returns `mixed` by default and PHPStan can't infer the returned type via Type Generics. So we need to tell it what we are turning here via PHPDoc types.